Functions/xASL_quant_Basil.m

```diff
@@ -123,170 +130,203 @@
 
 
 
-function [BasilOptions] = xASL_quant_Basil_Options(pathBasilOptions, x, PWI, bMultiPLD)
+function [BasilOptions] = xASL_quant_Basil_Options(pathBasilOptions, x, PWI)
 % Save a Basil options file and store CLI options for Basil
+% 1. Create an option file
+% 2. Basic tissue parameters
+% 3. Basic acquisition parameters
+% 4. Model fiting parameters
+% 5. Extra features on demand
+% 6. Save BASIL options file
+
+%% 1. Create option_file that contains options which are passed to Fabber
+% basil_options is a character array containing CLI args for the Basil command
+
+FIDoptionFile = fopen(pathBasilOptions, 'w+');
+BasilOptions = '';
+
+fprintf(FIDoptionFile, '# Basil options written by ExploreASL\n');
+fprintf(FIDoptionFile, '--iaf=diff\n'); % as input is PWI
+
+%% 2. Basic tissue parameters
+fprintf(FIDoptionFile, '--t1b=%f\n', x.Q.BloodT1/1000);
+fprintf(FIDoptionFile, '--t1=%f\n', x.Q.TissueT1/1000);
+
+%% 3. Basic acquisition parameters
+
+% Labelling type - PASL or pCASL
+switch lower(x.Q.LabelingType)
+	case 'pasl'
+		fprintf('BASIL: PASL model\n');
+		TIs = (unique(x.Q.Initial_PLD))'/1000;
+
+		% Print all the TIs
+		if x.modules.asl.bMultiPLD
+			% For Time-encoded, we skip the first volume
+			if x.modules.asl.bTimeEncoded
+				TIs = TIs(2:end);
+			end
+			for iTI = 1:length(TIs)
+				fprintf(FIDoptionFile, '--ti%d=%.2f\n', iTI, TIs(iTI));
+			end
+			
+		else
+			fprintf(FIDoptionFile, '--ti=%.2f\n', TIs);
+		end
+		
+		% Either print bolus duration or unspecify it
```

Comment:
Is this going correctly for multiple LDs? Or are we improving this later?

Reply:
This is about the code below !

Reply:
Good point. For PASL, the labeling duration is not defined by its actual duration (cause that is almost instantaneous) but by the timing of the bolus saturation pulse. And I guess that this doesn't really differ for multi-PLD sequences - or at least we don't really have the example for this. Still a valid point and I'll add a warning for future possible cases! Fixed here: 4dffd3ca